GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,120 @@
+                }
+            }
+
+            return count;
+        }
+
+        protected override void PerformMaintenance()
+        {
+            using (ITracer activity = this.Context.Tracer.StartActivity("LooseObjectsStep", EventLevel.Informational, Keywords.Telemetry, metadata: null))
+            {
+                try
+                {
+                    if (!this.ignoreTimeRestriction && !this.EnoughTimeBetweenRuns())
+                    {
+                        activity.RelatedWarning("Skipping step due to not enough time between runs");
+                        return;
+                    }
+
+                    // Verify no git processes are running
+                    IEnumerable<int> processIds = this.RunningGitProcessIds();
+                    if (processIds.Count() > 0)
+                    {
+                        activity.RelatedWarning("Skipping LooseObjectsStep due to git pids {0}", string.Join(",", processIds));
+                        return;
+                    }
+
+                    // Perform prune-packed
+                    int beforeCount = this.CountLooseObjects();
+                    GitProcess.Result result = this.RunGitCommand((process) => process.PrunePacked(this.Context.Enlistment.GitObjectsRoot));
+                    int afterCount = this.CountLooseObjects();
+
+                    // Record Telemetry
+                    EventMetadata metadata = new EventMetadata();
+                    metadata.Add("StartingCount", beforeCount);
+                    metadata.Add("EndingCount", afterCount);
+                    metadata.Add("RemovedCount", afterCount - beforeCount);
+                    activity.RelatedEvent(EventLevel.Informational, "LooseObjects", metadata, Keywords.Telemetry);
+
+                    // Update the history file
+                    this.RecordLooseObjectHistory();
+                }
+                catch (Exception e)
+                {
+                    activity.RelatedError("Failed to run LooseObjectsStep: {0}", e.ToString());
+                }
+            }
+        }
+
+        // Job should be run 7 days apart
+        private bool EnoughTimeBetweenRuns()
+        {
+            if (!this.Context.FileSystem.FileExists(this.looseObjectsHistoryPath))
+            {
+                return true;
+            }
+
+            string lastRunTime = this.Context.FileSystem.ReadAllText(this.looseObjectsHistoryPath);
+            DateTime dateTime = Convert.ToDateTime(lastRunTime);
+            double daysDiff = (DateTime.Now - dateTime).TotalDays;
+            if (daysDiff >= 7)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private IEnumerable<int> RunningGitProcessIds()
+        {
+            Process[] allProcesses = Process.GetProcesses();
+            return allProcesses.Where(x => x.ProcessName == "git").Select(x => x.Id);

```

Comment:
Good question!  

Since it's the name 'git' works.  InstallerPreRunChecker also has similar logic. 

Is git-checkout.exe a child process?  The installer also doesn't check for it.